File: 6/p099.py
import sys

sys.setrecursionlimit(100000)

def dfs(current):
    child[current] = 1
    res = 0
    for v in al[current]:
        if child[v] != -1:
            continue
        res += dfs(v)
        child[current] += child[v]
        res += child[v] * (n - child[v])
    return res

n = int(input())
al = [list() for _ in range(n)]
for i in range(n-1):
    u, v = map(int, input().split())
    al[u-1].append(v-1)
    al[v-1].append(u-1)

ans = 0
child = [-1]*n
ans = dfs(0)
print(ans)

File: 6/p099.go
package main

import (
	"bufio"
	"fmt"
	"os"
	"strconv"
)

var scan = bufio.NewScanner(os.Stdin)

const (
	initBufSize = 10000
	maxBufSize = 10000000
)

func init() {
	buf := make([]byte, initBufSize)
	scan.Buffer(buf, maxBufSize)
	scan.Split(bufio.ScanWords)
}

var n int
var visited []bool
var dp []int
var G [][]int

func main() {
	n = NumStdin()
	G = make([][]int, n)
	for i := 0; i < n-1; i++ {
		a, b := NumStdin(), NumStdin()
		G[a-1] = append(G[a-1], b-1)
		G[b-1] = append(G[b-1], a-1)
	}
	visited = make([]bool, n)
	dp = make([]int, n)

	dfs(0)

	ans := 0
	for i := 0; i < n; i++ {
		ans += dp[i] * (n - dp[i])
	}

	fmt.Println(ans)
}

func NumStdin() int {
	scan.Scan()
	num, err := strconv.Atoi(scan.Text())
	if err != nil {
		panic(err)
	}
	return num
}

func dfs(u int) {
	visited[u] = true
	dp[u] = 1
	for _, v := range G[u] {
		if visited[v] {
			continue
		}
		dfs(v)
		dp[u] += dp[v]
	}
}
